### DISCUSSION TITLE
Different defineable points

### DISCUSSION THREAD

**Discussion by Switch-Witch** [terjeio/ioSender#410] [Q&A]
Hi, some years ago I had a mill running a DSP controller (Richauto A11). It is a very basic but reliable system but there was one function wich was extremly usefull: You was able to define different 0 Points for X, Y, Z.  Espacially when you have different vises or/and a fourth axis attached it is very frustrating to search for every desired location again and again. For sure this is possible to realize using macros but it would be very handy to define for example the number keys of the keyboard with the different points. Ist there any function I did not find yet or would it be possible to implement it? best Kai

**Comment by terjeio** [comment]
grblHAL supports offsets [G54-G59.3](http://linuxcnc.org/docs/html/gcode/g-code.html#gcode:g54-g59.3), legacy Grbl G54-G59. You can create macros for switching between them, or just send the G-code from the MDI. Use the _Offset_ tab to set them.

**Reply by Switch-Witch** [reply]
Thx so much for your help!
That`s what I was searching for.

best

Kai
